Clean up more lint in toolbox __init__.py.

gerph opened the pull request:
This change reduces the lint errors significantly, whilst keeping the wildcard imports. This is achieved by ignoring some errors and making some of the code use slightly different ways of importing the code.

gerph committed:
Rework the fakeswi to use class methods.

The fakeswi interface is a little more flexible and can patch multiple
modules if it uses a class instead of the functions being bare inside
the fakeswi module.

This might not be quite so useful in general, but it seemed more
flexible to me.

gerph committed:
Remove redundant commented out mocking code from fakeswi.

The fakeswi module had commented out mocking code present, which has
been duplicated in saveas. In the future it might be moved to the
fakeswi module, but right now it's not necessary.

gerph committed:
Allow runtest.sh to take parameters.

runtest.sh now allows parameters to be passed, so we can run
`runtest.sh -v` to get more useful output.

gerph committed:
Clean up more lint in toolbox __init__.py.

This change reduces the lint errors significantly, reducing it to
just the base imports. Instead of using the wildcard imports, we
now import only the classes, methods and functions that we use.

This might cause some clients to break as they assume that the top
level module contains constants, but it is far cleaner. It might be
that instead of wildcard we need to specify everything, or
alternatively use __all__ in the imported modues, instead of this.

c-jo commented:
It might be we need to use __all__, at least for now as client code will be making the assumtion it can import things from riscos_toolbox. My feeling is they should probably be importing them from riscos_toolbox.something (eg. from riscos_toolbox.point import Point) but that might be a breaking change too far for now.

I think importing them directly gives a different linter message, along the lines of "You imported this but aren't using it"

gerph commented:
I think you're right; I wasn't too keen on this change because it might break clients that assumed they could import from `riscos_toolbox`. I'll have another look at it later. Don't worry for now.

gerph committed:
Restore back to using 'import *' with warnings for flake8.

The 'import *' is generally unwise, but has been used here to make
it easier for clients to import objects.

gerph committed:
Merge remote-tracking branch 'fork/rework-fakeswi' into clean-more-lint-toolbox-init

gerph committed:
Clean up the fakeswi imports.

The fakeswi imports are now a bit smaller and easier to manage.
Instead of doing conditional things for non-RISC OS, the fakeswi
is used everywhere to ensure that we don't import that SWI interfaces
on RISC OS and have managed interfaces on non-RISC OS systems.

gerph commented:
Was 46 failures in `flake8.sh`; now down to 6..
